### PR TITLE
Verifier Sourcify service implementation

### PR DESCRIPTION
**Pull request by rimrakhimov**
User story: https://github.com/blockscout/blockscout-rs/issues/174

Adds Sourcify service implementation for Grpc/Http smart-contract-verifier server.

Depends on https://github.com/blockscout/blockscout-rs/pull/225

### CODE REVIEW

File: smart-contract-verifier/smart-contract-verifier-server/src/services/sourcify_verifier.rs [hunk 1]
```diff
@@ -1,17 +1,52 @@
...
+}
+
+impl SourcifyVerifierService {
+    pub fn new(settings: SourcifySettings) -> anyhow::Result<Self> {
+        let client = SourcifyApiClient::new(
+            settings.api_url,
+            settings.request_timeout,
+            settings.verification_attempts,
+        )
+        .expect("failed to build sourcify client");
```
Comment (leviska) [inline smart-contract-verifier/smart-contract-verifier-server/src/services/sourcify_verifier.rs:24]: no extensions

File: smart-contract-verifier/smart-contract-verifier-server/src/types/sourcify.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+use serde::{Deserialize, Serialize};
+use smart_contract_verifier::sourcify::api::VerificationRequest;
+use smart_contract_verifier_proto::blockscout::smart_contract_verifier::v1::VerifyViaSourcifyRequest;
+use std::ops::Deref;
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct VerifyViaSourcifyRequestWrapper(VerifyViaSourcifyRequest);
```
Comment (leviska) [inline smart-contract-verifier/smart-contract-verifier-server/src/types/sourcify.rs:7]: couldn't we just make a `fn convert_sourcify(VerifyViaSourcifyRequest) -> VerificationRequest` and have like 10% of the code? :)

File: smart-contract-verifier/smart-contract-verifier-server/src/types/verification_response.rs [hunk 3]
```diff
@@ -126,6 +127,32 @@ pub mod verify_response {
         }
     }
 
+    impl From<SourcifySuccess> for ResultWrapper {
```
Comment (leviska) [inline smart-contract-verifier/smart-contract-verifier-server/src/types/verification_response.rs:130]: may be we should write some macro for this later

File: smart-contract-verifier/smart-contract-verifier-server/src/types/sourcify.rs (hunk 2 above)
Comment (rimrakhimov) [reply smart-contract-verifier/smart-contract-verifier-server/src/types/sourcify.rs:7]: We canRead https://github.com/blockscout/blockscout-rs/pull/224#discussion_r1034041322 for more details

File: smart-contract-verifier/smart-contract-verifier-server/src/services/sourcify_verifier.rs (hunk 1 above)
Comment (rimrakhimov) [reply smart-contract-verifier/smart-contract-verifier-server/src/services/sourcify_verifier.rs]: Added in https://github.com/blockscout/blockscout-rs/pull/226/commits/e3652d887b34462822b79b9e7739ade9a9277736

**Review by sevenzing — APPROVED** [review]

**Pull request merged**